bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -2642,21 +2650,46 @@ private static void beginTransaction(WorkerExecutionContext ctx, int transaction
                 return;
             }
         }
+
+        //Register committed function handler if exists.
+        if (committedFuncIndex != -1) {
+            FunctionRefCPEntry funcRefCPEntry = (FunctionRefCPEntry) ctx.constPool[committedFuncIndex];
+            BFunctionPointer fpCommitted = new BFunctionPointer(funcRefCPEntry);
+            TransactionResourceManager.getInstance().registerCommittedFunction(transactionBlockId, fpCommitted);
+        }
+
+        //Register aborted function handler if exists.
+        if (abortedFuncIndex != -1) {
+            FunctionRefCPEntry funcRefCPEntry = (FunctionRefCPEntry) ctx.constPool[abortedFuncIndex];
+            BFunctionPointer fpAborted = new BFunctionPointer(funcRefCPEntry);
+            TransactionResourceManager.getInstance().registerAbortedFunction(transactionBlockId, fpAborted);
+        }
+
         LocalTransactionInfo localTransactionInfo = ctx.getLocalTransactionInfo();
         boolean isInitiator = true;
         if (localTransactionInfo == null) {
-            BValue[] returns = notifyTransactionBegin(ctx, null,  null, transactionBlockId,
-                            TransactionConstants.DEFAULT_COORDINATION_TYPE, isInitiator);
-            BStruct txDataStruct = (BStruct) returns[0];
-            String globalTransactionId = txDataStruct.getStringField(1);
-            String protocol = txDataStruct.getStringField(2);
-            String url = txDataStruct.getStringField(3);
+            String globalTransactionId = "";
+            String protocol = "";
```

Comment:
Why are these variables initialized with empty strings? .. shouldn't they be null if they are not assigned a value.